internal/artifacts/nighttimewhispersintheechoingwoods/nighttimewhispersintheechoingwoods.go

```diff
@@ -0,0 +1,127 @@
+package nighttimewhispersintheechoingwoods
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/info"
+	"github.com/genshinsim/gcsim/pkg/core/keys"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/core/player/shield"
+	"github.com/genshinsim/gcsim/pkg/modifier"
+)
+
+func init() {
+	core.RegisterSetFunc(keys.NighttimeWhispersInTheEchoingWoods, NewSet)
+}
+
+type Set struct {
+	Index int
+	core  *core.Core
+	char  *character.CharWrapper
+	lastF int
+}
+
+func (s *Set) SetIndex(idx int) { s.Index = idx }
+func (s *Set) Init() error      { return nil }
+
+func NewSet(c *core.Core, char *character.CharWrapper, count int, param map[string]int) (info.Set, error) {
+	s := Set{
+		core:  c,
+		char:  char,
+		lastF: 0,
+	}
+
+	if count >= 2 {
+		m := make([]float64, attributes.EndStatType)
+		m[attributes.ATKP] = 0.18
+		char.AddStatMod(character.StatMod{
+			Base:         modifier.NewBase("nwew-2pc", -1),
+			AffectedStat: attributes.ATKP,
+			Amount: func() ([]float64, bool) {
+				return m, true
+			},
+		})
+	}
+
+	if count >= 4 {
+		c.Events.Subscribe(event.OnShielded, OnShielded(&s), fmt.Sprintf("nwew-4pc-%v", char.Base.Key.String()))
+		c.Events.Subscribe(event.OnShieldBreak, OnShieldBreak(&s), fmt.Sprintf("nwew-4pc-%v", char.Base.Key.String()))
+		c.Events.Subscribe(event.OnCharacterSwap, OnCharacterSwap(&s), fmt.Sprintf("nwew-4pc-%v", char.Base.Key.String()))
+		c.Events.Subscribe(event.OnSkill, OnSkill(&s), fmt.Sprintf("nwew-4pc-%v", char.Base.Key.String()))
+	}
+
+	return &s, nil
+}
+
+func OnShielded(s *Set) func(args ...interface{}) bool {
+	return func(args ...interface{}) bool {
+		shd := args[0].(shield.Shield)
+		if s.core.Player.Active() != s.char.Index {
+			return false
+		}
```

Comment:
This should be removed because what if the holder is off-field when the on-field character gains the crystallize shield? When swapping back onto the holder they will not benefit from the *2.5 increase even though they are protected by a crystallize shield.

Reply:
Nvm it is covered by the swap sub

Reply:
Yep